move weekly digest email to sidekiq-cron and remove whenever

markets opened the pull request:
Closes #373 
Closes https://github.com/coopdevs/timeoverflow-provisioning/issues/24
Closes #544
Closes #548 

**EXTRA:**
- update more dev deps: rubocop, byebug, fabrication, capybara, webdrivers
- fixing CI (feature specs)

**INFRA CHANGES & TESTING**
👀 https://github.com/coopdevs/timeoverflow/pull/401#issuecomment-471716708

enricostano commented:
To test this we first need to solve https://github.com/coopdevs/timeoverflow-provisioning/pull/36

markets commented:
Ok @enricostano thanks! I think this will also close #369

sauloperez commented on app/jobs/organization_notifier_job.rb:
```diff
@@ -0,0 +1,12 @@
+class OrganizationNotifierJob < ActiveJob::Base
```
It'd be great to explain what it notifies about and who without having to read the code. I'd be much faster for readers to better understand what it does.

enricostano commented:
Cool @markets , many thanks for improving this PR! :muscle: 

Just two questions:

1. What would be the best strategy to test it?
2. I guess we should plan some kind of pre-post deploy operations for production to remove the existent cron job, right?

markets commented:
Hi @enricostano 

1) the code inside the Job class is just a copy/pasta from the old Service class (with `find_each` instead of `each`), so in terms of logic, it should work exactly as it works now. To ensure that, I think we can use the Sidekiq UI, as cron jobs are displayed there and they have an "Enqueue Now" option:

![Captura de pantalla 2019-03-11 a la(s) 21 21 32](https://user-images.githubusercontent.com/576701/54155116-c3ccfe80-4443-11e9-882b-ea5ebd9521f8.png)

2) We should remove the crontab line generated by the `whenever` gem, this can be done by hand (`crontab -e`) or using an option provided by the gem: `whenever -c`

sseerrggii commented:
:thinking: I think I will need som help to test that @enricostano also I think we have som troubles with push notifications, so one day we can sit together and test thinks looking at  Sidekiq :pray:

enricostano commented:
Blocked by https://github.com/coopdevs/timeoverflow-provisioning/pull/36

markets commented:
Just updated with current develop! I also "unblocked" this PR, as the main blocker https://github.com/coopdevs/timeoverflow-provisioning/pull/36 was already solved.

sseerrggii commented:
Testing it on staging i get this error

![FireShot Capture 377 -  STAGING  Sidekiq - staging timeoverflow org](https://user-images.githubusercontent.com/2297137/74964091-11ebcf80-5413-11ea-88d9-18b879c654f9.png)

markets commented:
Yay! Travis is passing ✅ now! It's already failing in master, and not related to this branch, but I fixed it here in order to have green build for this branch.

@sseerrggii I also fixed the error you got last day. Thanks!

sauloperez commented:
Thanks a ton @markets ! what a mess with the locale :see_no_evil: @mperezv this is relevant to you :point_up:

sseerrggii commented:
Now Works fine, thanks @markets :green_heart:

sauloperez commented:
Finally up and working in production! thanks a lot @markets :tada:  ! TO is now cron-free and relies on Sidekiq for all kinds of async stuff :clap:
